File: src/p1_two_sum.rb
# frozen_string_literal: true

#
# @lc app=leetcode id=1 lang=ruby
#
# [1] Two Sum
#

# @lc code=start
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
  record = {}

  nums.each_with_index do |num, i|
    return [record[num], i] if record.include?(num)

    record[target - num] = i
  end

  []
end
# @lc code=end

print(two_sum([2, 7, 11, 15], 9))

File: src/p1.twoSum.php
<?php
/*
 * @lc app=leetcode id=1 lang=php
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution
{
    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target)
    {
        $record = [];

        foreach ($nums as $i => $num) {
            if (array_key_exists($num, $record)) {
                return [$record[$num], $i];
            }

            $record[$target - $num] = $i;
        }

        return [];
    }
}
// @lc code=end


echo json_encode((new Solution())->twoSum([2, 7, 11, 15], 9));
